File: Codes/C++/hellov2.cpp
#include <iostream>
#include <string>
#include <chrono>
#include <thread>

/*fazendo hello world de uma forma diferente*/

class HelloWorld{

  private:
     std::string ola_mundo{"Hello world!"};

  public:

    //construtor
    HelloWorld(void){
      std::cout << "Inicializando objHelloWorld..." << std::endl;
      std::this_thread::sleep_for(std::chrono::seconds(1));
    }

    //destrutor
    ~HelloWorld(void){
      std::cout << "Desligando objHelloWorld..." << std::endl;
      std::this_thread::sleep_for(std::chrono::seconds(1));
    }

    //método que imprime ola_mundo
    void printfworld(void){
      std::cout << "Imprimindo..." << std::endl;
      std::this_thread::sleep_for(std::chrono::seconds(1));
      std::cout << ola_mundo << std::endl;
    }

}; 

int main ( void ){

  HelloWorld hello;

  hello.printfworld();

  return 0;
  
}

File: Codes/Python/hellov2.py
# fazendo a mesma coisa do hellov2.cpp, fazer um hello world com Orientação a Objetos


import time as tm

class HelloWorld:

    #construtor

    def __init__ (self):
        print('Inicializando o objeto que vai falar algo...')
        tm.sleep(2)

    #destrutor

    def __del__ (self):
        print('Destruindo o objeto :(')
        tm.sleep(2)


    def falar(self):
        print('\nHello World!\n')

def main():

    hello = HelloWorld # o objeto nasce aqui e aparece o __init__ 

    hello.falar() # roda o método

    del hello #destrutor roda aqui

if __name__ == "__main__":
    main()
